noah_frontend/src/components/donate.js

```diff
@@ -84,7 +89,8 @@ class DonateView extends React.Component {
         let errors = {}
         const phoneRegex = /^\d{10}$/,
               pinCodeRegex = /^\d{6}$/,
-              landPhoneRegex = /^0\d{10}$/
+              landPhoneRegex = /^0\d{10}$/,
+              nameRegex = /^[a-zA-Z][-_ a-zA-Z0-9]+$/
         if (!(phoneRegex.test(donationForm['contact_number'].value) || landPhoneRegex.test(donationForm['contact_number'].value))) {
           errors.contact_number = 'Enter 10 digit mobile number or 11 digit landline number'
         } else if(formErr.contact_number){
```

Comment:
Why is this condition left unhandled?

Reply:
if the error was present for contact_number in UI, and currently the error has resolved, then updating the state (in line number 100)

Reply:
remove the condition if not used

Reply:
We are using it. The state is updating and rendering error based on that